### PR TITLE
[ID-398] Avoid always showing ToS page when loading Terra

### PR DESCRIPTION
**Pull request by nawatts**
Currently, when loading Terra, the ToS page is always briefly shown. It The effects of this are visible as...
- a flicker in the loading spinner
- a React state warning in the console (because the ToS page doesn't cancel its request for ToS text when unmounted)
- the request for the ToS text in the network inspector.

https://user-images.githubusercontent.com/1156625/212907326-069324bf-a1f2-4bdc-b99f-5a7c9645e22d.mov

---

AuthContainer displays the ToS page if the user can't use Terra or if they have not continued under a ToS grace period.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/components/AuthContainer.js#L19

However, when it comes to the grace period, it actually only shows the ToS page if `termsOfService.userContinuedUnderGracePeriod` is undefined.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/components/AuthContainer.js#L29

When a user first loads Terra or first signs in, `termsOfService.userContinuedUnderGracePeriod` will be undefined.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/libs/auth.js#L250-L259

Thus, AuthContainer will always show the ToS page.

However, when mounted, the ToS page sets `termsOfService.userContinuedUnderGracePeriod` to false.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/pages/TermsOfService.js#L27-L33

That makes AuthContainer's condition for showing the ToS page false, and AuthContainer renders the page for the current route. Since `termsOfService.userContinuedUnderGracePeriod` is not persisted anywhere, this occurs every time Terra is loaded.

Because of this, `termsOfService.userContinuedUnderGracePeriod` is effectively useless. It's never used in a way that will make the ToS page visible for long enough for the user to actually see it. And so, this PR removes it.

---

However, that alone doesn't fix the issue. AuthContainer displays the ToS page when `termsOfService.userCanUseTerra` is falsy.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/components/AuthContainer.js#L19

When a user loads Terra or signs in, `termsOfService.userCanUseTerra` is initialized to undefined.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/libs/auth.js#L250-L259

Terra UI then fetches the user's ToS status from Sam and sets `termsOfService.userCanUseTerra` based on that.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/libs/auth.js#L331-L337

But while that request is in progress, `termsOfService.userCanUseTerra` will be falsy and the ToS page will be shown. Once the request completes, if the user has accepted the ToS, the ToS page will be removed.

Most of the time, when a signed in user is loading Terra, it's going to be a user that has already accepted the ToS as opposed to a new user that needs to. By changing `!termsOfService.userCanUseTerra` to `termsOfService.userCanUseTerra === false` in AuthContainer's logic for displaying the ToS page, this waits until the user's ToS status has been loaded from Sam to show the ToS page.

### CODE REVIEW

File: integration-tests/tests/register-user.js
Comment (nawatts) [inline integration-tests/tests/register-user.js:17]: For some reason, this test was ending up on the workspaces page. But after clicking the accept button, Terra redirects to the home page.

https://github.com/DataBiosphere/terra-ui/blob/c0d416875ed08db77082d3f64393f0e783089d49/src/pages/TermsOfService.js#L58

File: src/pages/TermsOfService.js
Comment (nawatts) [inline src/pages/TermsOfService.js:69]: I suspect this was originally copied from the `accept` function. While `accept` contains an API request, there's nothing here that requires the busy state or error handling.

File: integration-tests/tests/register-user.js
Comment (cahrens) [reply integration-tests/tests/register-user.js:17]: I believe it should end up on the workspaces pages because the log in process was initiated by clicking "View Workspaces" (line 10). So after registering, it should remember to go back to Workspaces.

Comment (nawatts) [reply integration-tests/tests/register-user.js:17]: That was true before #3671, but that PR changed the behavior of the accept button from `Nav.history.goBack()` to `Nav.goToPath('root')`.

https://github.com/DataBiosphere/terra-ui/pull/3671/files#diff-6d4c9b0f2ed976516728fc98d614864ed93e78b692554d0e6a802d03fd60c890L49-R58

Comment (cahrens) [reply integration-tests/tests/register-user.js:17]: Why was that change made?

Comment (nawatts) [reply integration-tests/tests/register-user.js:17]: @gpolumbo-broad @Ghost-in-a-Jar can you answer this question?

Comment (Ghost-in-a-Jar) [reply integration-tests/tests/register-user.js:17]: When testing this functionality across different environments we noticed that goBack would send us to whatever url we were browsing before opening terra. Sending us out of the platform. 

The rationale was to send users back to root instead of potentially outside of the platform. We are also react noobs though so I'm sure there is a better way to achieve the UX we want.

**Review by Ghost-in-a-Jar — APPROVED** [review]
We were in a rush to get a band aid in Friday so I am not surprised we fudged this. Thanks again @nawatts for being proactive about fixing!

File: integration-tests/tests/register-user.js
Comment (cahrens) [reply integration-tests/tests/register-user.js:17]: I think it would definitely be better if `goBack` was used as long as it doesn't take you outside of Terra. I don't know off the top of my head how to do that.

But I'm curious-- what URL were you starting with that immediately required you to register? `https://app.terra.bio/` doesn't make you register immediately, so presumably one needs to click on something else in Terra to initiate the registration process.

**Review by gpolumbo-broad — APPROVED** [review]

File: integration-tests/tests/register-user.js
Comment (Ghost-in-a-Jar) [reply integration-tests/tests/register-user.js:17]: @cahrens this was behavior we noticed when running locally and in dev in particular.

I would go to the landing page, login, be shown tos, then get sent back to the previous URL I was at (before the terra landing page).

Comment (cahrens) [reply integration-tests/tests/register-user.js:17]: Oh, perhaps because you clicked log in vs. clicking on a page link that forced the log in.

Comment (Ghost-in-a-Jar) [reply integration-tests/tests/register-user.js:17]: yeah that is a possibility -- i would definitely defer to UI folks though because Greg and I are total react noobs.

Comment (Ghost-in-a-Jar) [reply integration-tests/tests/register-user.js:17]: I think we could probably code something fancy here as a followup if we want. Like use goBack to the last terra page in the session history -- if there arent any terra pages in the history just go to root.

Comment (nawatts) [reply integration-tests/tests/register-user.js:17]: Agreed. In any case, `toToPath('root')` is currently in production, so I think changes to that can be a follow-up.

**Pull request merged**
